### PR TITLE
Speaker - part2

### PR DESCRIPTION
**Pull request by Jungjee**
make trainable
- currently WIP

### CODE REVIEW

File: espnet2/spk/espnet_model.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+from espnet2.layers.abs_normalize import AbsNormalize
+from espnet2.torch_utils.device_funcs import force_gatherable
+from espnet2.train.abs_espnet_model import AbsESPnetModel
+from espnet.nets.pytorch_backend.nets_utils import to_device
+from espnet2.spk.loss import *
+
+if V(torch.__version__) >= V("1.6.0"):
+    from torch.cuda.amp import autocast
+else:
+    # Nothing to do if torch<1.6.0
```
Comment (Emrys365) [inline espnet2/spk/espnet_model.py:30]: You don't need this check since the supported PyTorch versions in ESPnet are now all higher than 1.6.0.

File: espnet2/spk/loss/aamsoftmax.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+# -*- encoding: utf-8 -*-
+# code from https://github.com/clovaai/voxceleb_trainer/blob/master/loss/aamsoftmax.py
+# Adapted from https://github.com/wujiyang/Face_Pytorch (Apache License)
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import time, pdb, numpy, math
+
+class LossFunction(nn.Module):
```
Comment (Emrys365) [inline espnet2/spk/loss/aamsoftmax.py]: It might be better to define a common abstract Loss class if you want to add more losses later.

File: espnet2/spk/encoder/rawnet3_encoder.py
Comment (Emrys365) [inline espnet2/spk/encoder/rawnet3_encoder.py:1]: Note that you need to add an empty `__init__.py` file in each new subdirectory under `espnet2`, such that they can be imported as a module.
